api/v1beta1/common_types.go

```diff
@@ -57,6 +57,14 @@ const (
 	// update that disallows a pre-existing Cluster to be populated with Topology information and Class.
 	ClusterTopologyUnsafeUpdateClassNameAnnotation = "unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check"
 
+	// MachinePoolOwnedLabel is the label set on Machines and InfraMachines linked to a MachinePool.
+	MachinePoolOwnedLabel = "machinepool.cluster.x-k8s.io/owned"
```

Comment:
I think in this case we probably can since this label is only really used in the machine webhook. I chose to rely on labels for the most part because it allows us to query Machines quickly with `client.List()` and imo it's a bit cleaner to do a key value look up rather than searching through a list.